core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -755,6 +761,13 @@ oppia.directive('conversationSkin', [
           };
 
           $scope.showUpcomingCard = function() {
+            /* This is because when answerIsCorrect becomes false below, if
+               pendingCardWasSeenBefore is true (going forward to the card from
+               which redirection just happened), Learn Again would be briefly
+               visible. */
+            if ($scope.answerIsCorrect) {
```

Comment:
Do you actually need the 'if' check?

Reply:
If the 'if' check was not there, I found that Learn Again button briefly switched to Continue before going to next card ( opposite of current issue)

Reply:
Ah ok. I would add a comment explaining that that's why the if check is necessary.

Reply:
Ok, will add that